### PR TITLE
Array manipulation 3

### PR DESCRIPTION
**Pull request by OSAdams**
image

### CODE REVIEW

File: array-manipulation-3/src/difference.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/* exported difference */
+function difference (first, second) {
+  return [
+    ...first.filter(x => !second.includes(x)),
+    ...second.filter(x => !first.includes(x))
+  ];
+}
```
Comment (thebearingedge) [inline array-manipulation-3/src/difference.js:7]: Please rewrite this without arrow functions, spread, or filter. It's very risky to turn in code you are not prepared to fully explain.

File: array-manipulation-3/src/zip.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      newArray.push(innerArrayTwo);
+    }
+  }
+  if (first.length === second.length) {
+    for (var k = 0; k < first.length; k++) {
+      var innerArrayThree = [];
+      innerArrayThree.push(first[k], second[k])
+      newArray.push(innerArrayThree);
+    }
+  }
```
Comment (thebearingedge) [inline array-manipulation-3/src/zip.js:24]: If you decide which array is shorter, you can write one loop.

```js
var length = Math.min(first.length, second.length);
```

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: array-manipulation-3/src/difference.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    }
+  }
+  for (var j = 0; j < length; j++) {
+    if (second[j] !== first[j]) {
+      newArray.push(second[j]);
+      console.log(second[j])
+    }
+  }
+  return newArray;
+}
```
Comment (thebearingedge) [inline array-manipulation-3/src/difference.js:18]: This function can be simplified a bit. Try this strategy:
- create a new empty array
- loop through the first array
- for each value
    - if the value is not in the second array, put it in the new array
- loop through the second array
- for each value
    - if the value is not in the first array, put it in the new array
- return the new array

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Comment by OSAdams** [comment]
image

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**